packages/flutter/lib/src/rendering/platform_view.dart

```diff
@@ -322,6 +322,9 @@ class RenderUiKitView extends RenderBox {
   UiKitViewController _viewController;
   set viewController(UiKitViewController viewController) {
     assert(viewController != null);
+    if (_viewController == viewController) {
+      return;
+    }
     final bool needsSemanticsUpdate = _viewController.id != viewController.id;
     _viewController = viewController;
```

Comment:
This is confusing code, because the local `viewController` is shadowing the getter `viewController`. Perhaps while you are here you could rename the parameter to something harmless like `value`?

Reply:
Done (in latest commit). 

Btw this seems also commonly seen in flutter framework code (though I agree `value` is neater):

![image](https://user-images.githubusercontent.com/5236035/190832094-d14a77cd-cbc3-49b8-b8d2-77454111c531.png)